dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -1823,6 +1823,19 @@ public void run() {
                 return output;
             }
         }
+        if (failedRestore()) {
+            // Did not find restore message; assume it failed;
+            // The return code is 0 indicating the server started, likely recovered
+            // set as started but then stop the server
+            setStarted();
+            try {
+                stopServer();
+            } catch (Exception e) {
+                Log.error(c, method, e);
+                // we don't want to fail if stop fails
+            }
+            fail("The server did not restore successfully");
```

Comment:
I wonder if failing with an assert like this will still allow for collection of the server log to diagnose the restore from checkpoint which failed? It's similar to the fail assertion for `if (output.getReturnCode() != 0)` above but I don't recall seeing it fail in that code path, so not sure if it will always collect logs there either. We _may_ need a `postStopServerArchive()` call